kubernetes/samples/scripts/common/validate.sh

```diff
@@ -0,0 +1,335 @@
+# $1 - the output directory to validate
+# $2-n - the names of the generated yaml files
+function internalValidateGeneratedYamlFilesDoNotExist {
+  local dir=$1
+  shift
+  for var in "$@"; do
+    local f="${dir}/${var}"
+    if [ -e ${f} ]; then
+      validationError "${f} exists."
+    fi
+  done
+}
+
+#
+# Function to validate the version of the inputs file
+#
+function validateVersion {
+  local requiredVersion=${requiredInputsVersion}
+  if [ "${version}" != "${requiredVersion}" ]; then
+    validationError "Invalid version: \"${version}\".  Must be ${requiredVersion}."
+  fi
+}
+
+#
+# Function to ensure the domain uid is a legal DNS name
+#
+function validateDomainUid {
+  validateLowerCase "domainUID" ${domainUID}
+  validateDNS1123LegalName domainUID ${domainUID}
+}
+
+#
+# Function to ensure the namespace is lowercase
+#
+function validateNamespace {
+  validateLowerCase "namespace" ${namespace}
+}
+
+#
+# Create an instance of clusterName to be used in cases where a legal DNS name is required.
+#
+function validateClusterName {
+  clusterNameSVC=$(toDNS1123Legal $clusterName)
+}
+
+#
+# Create an instance of adminServerName to be used in cases where a legal DNS name is required.
+#
+function validateAdminServerName {
+  adminServerNameSVC=$(toDNS1123Legal $adminServerName)
+}
+
+#
+# Create an instance of adminServerName to be used in cases where a legal DNS name is required.
+#
+function validateManagedServerNameBase {
+  managedServerNameBaseSVC=$(toDNS1123Legal $managedServerNameBase)
+}
+
+#
+# Function to validate the secret name
+#
+function validateWeblogicCredentialsSecretName {
+  validateLowerCase "weblogicCredentialsSecretName" ${weblogicCredentialsSecretName}
+}
+
+# try to execute kubectl to see whether kubectl is available
+function validateKubectlAvailable {
+  if ! [ -x "$(command -v kubectl)" ]; then
+    validationError "kubectl is not installed"
+  fi
+}
+
+# Function to validate the server startup control value
+#
+function validateStartupControl {
+  validateInputParamsSpecified startupControl
+  if [ ! -z "${startupControl}" ]; then
+    case ${startupControl} in
+      "NONE")
+      ;;
+      "ALL")
+      ;;
+      "ADMIN")
+      ;;
+      "SPECIFIED")
+      ;;
+      "AUTO")
+      ;;
+      *)
+        validationError "Invalid value for startupControl: ${startupControl}. Valid values are 'NONE', 'ALL', 'ADMIN', 'SPECIFIED', and 'AUTO'."
+      ;;
+    esac
+  fi
+}
+
+#
+# Function to validate the cluster type value
+#
+function validateClusterType {
+  validateInputParamsSpecified clusterType
+  if [ ! -z "${clusterType}" ]; then
+    case ${clusterType} in
+      "CONFIGURED")
+      ;;
+      "DYNAMIC")
+      ;;
+      *)
+        validationError "Invalid value for clusterType: ${clusterType}. Valid values are 'CONFIGURED' and 'DYNAMIC'."
+      ;;
+    esac
+  fi
+}
+
+#
+# Function to validate the weblogic domain storage reclaim policy
+#
+function validateWeblogicDomainStorageReclaimPolicy {
```

Comment:
I thought the customer was going to configure the PV & PVC, not the sample, therefore the inputs file wouldn't contain the properties needed to configure them.